examples/HowTo.md

```diff
@@ -0,0 +1,147 @@
+
+So to make it possible we should transformed this function to the following:
+
+```haskell
+mainWithLogging :: IO ()
+mainWithLogging = do
+    buildAndSetupYamlLogging productionB "examples/how-to-log-config.yaml"
+    usingLoggerName "new-logger" inputLengthWithLog
+```
+where we set up the config from the file and run `inputLengthWithLog` under the logger with name `new-logger`.
+
+Let's have a closer look at how we transform the `inputLength` function to the `inputLengthWithLog` that would work with logging in the next paragraph.
+
+## Add logging messages to function in `IO`
+
+If you want to use logging in a function that already runs in the `IO` monad, replace the monad with [`WithLoggerIO`](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#t:WithLoggerIO) and `liftIO` all your `IO` actions.
+
+For example, our `inputLength` function works this way:
+```haskell
+
+inputLength :: IO ()
+inputLength = do
+    input <- TIO.getLine
+    case T.length input of
+        0 -> putStrLn "Should not be empty" >> inputLength
+          -- v ^  you need to log these
+        _ -> putStrLn "Well done" >> funWithInput input
+
+```
+Very easy, nothing interesting. We have some input from console and write some messages depending of the length of user's input.
+What we are going to do is to log this messages instead of just `putStrLn` them.
+
+First that we should consider is that logging can not be done in `IO`, so we need make the function work in `WithLoggerIO` monad,
+so that we are able to do any logging stuff inside.
+After this is done you can easily use `logDebug`, `logInfo`, `logWarning` etc.
+(see others [here](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#v:logDebug)) to log messages.
+As it's not `IO` after our transformations, we shouldn't forget to `liftIO` functions that work in `IO` and doesn't have logging inside.
+As result we will get
+
+```haskell
+
+inputLengthWithLog :: WithLoggerIO m => m ()
+inputLengthWithLog = do
+    input <- liftIO TIO.getLine
+    case T.length input of
+        0 -> do
+            logError "Should not be empty"
+            inputLengthWithLog
+        _ -> do
+            logInfo "Well done"
+            funWithInputInMonad input
+
+```
+
+Another moment, let's have a closer look at the `funWithInput` function from the example above.
+We see that its type is `IO ()` and it doesn't have logging inside.
+
+```haskell
+funWithInput :: Text -> IO ()
+funWithInput input = do
+    TIO.putStrLn $ "Wow! You wrote " <> input
+
+```
+So we also can rewrite it in this way
+```haskell
+funWithInputInMonad :: MonadIO m => Text -> m ()
+funWithInputInMonad input =
+    liftIO $ TIO.putStrLn $ "Wow! You wrote " <> input
+
+```
+or usage of `liftIO` is also the solution.
+
+And as a conclusion, let's write the `main` function to check how we managed to apply logging in very simple example
```

Comment:
I still want to see screenshot with the result of running `how-to` and observing results... Like in this way:

1. `stack exec how-to`
2. Input empty string
3. Input "abacaba"
4. Input empty string.
5. Input "abacaba"
6. Make screenshot

So user can see how messages are printed.